### ISSUE TITLE
Fecth request timing out on Vercel production

### ISSUE THREAD

**Issue by flaviobvds** [vercel/next.js#47101]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
I'm trying to fetch a specific page, but it times out and never gets the response back. The response is only 18KB and the fetch request works fine in dev on localhost. However when I deploy it to vercel, it simply doesn't fetch anymore. I've tried using axios, but same issue occurs.
This is the JSON I'm trying to fetch:
https://www.canada.ca/content/dam/ircc/documents/json/ee_rounds_123_en.json
```

### Link to the code that reproduces this issue

https://github.com/flaviobvds/eedraws/blob/main/src/pages/api/checkUpdates.ts

### To Reproduce

1 - Write a serverless function inside api routes folder
2 - Inside the function call the fetch method fetching the URL: https://github.com/flaviobvds/eedraws/blob/main/src/pages/api/checkUpdates.ts
3 - Deploy to production on Vercel
4 - It will timeout

### Describe the Bug

When I try to fetch the page `https://github.com/flaviobvds/eedraws/blob/main/src/pages/api/checkUpdates.ts` from a Vercel deploy it takes forever and never loads. Same thing happens if I try using axios. It works normally on localhost, though. Apparently it's something related to that page specifically, as I've tried fetching other JSONs from the web and it worked fine.

Next.JS v13.2.3

### Expected Behavior

It should return a response data, but instead it never does and ends up timing out.

### Which browser are you using? (if relevant)

Brave

### How are you deploying your application? (if relevant)

Vercel

**Comment by Pardaramskha** [comment]
I'm having kind of the same issue atm and [started a discussion around it](https://github.com/vercel/next.js/discussions/47237).
Though a pair of solutions have been tried here and there, none of them work by now.

@flaviobvds does your app throws you a 504 error ? Does it affect GSSP too ?

**Comment by flaviobvds** [comment]
@Pardaramskha I was able to workaround this by using `node-fetch` instead of `fetch` or `axios`.
Here's the code that worked for me:

``` javascript
import nodefetch from 'node-fetch'

const data = await nodefetch(url)
```

I still have no idea why I wasn't able to use either `fetch` or `axios`.
I asked it on [StackOverflow](https://stackoverflow.com/questions/75737513/next-js-fetch-and-axios-request-timing-out-on-vercel-production) but the only answer I got is that it probably means that Vercel was running Node.js 16, but that isn't the case for me, because mine is using Node.js 18, so I have no idea of what's going on, especially because I was able to use `fetch` in other parts of my code, but I was able to get around it by using node-fetch

By the way, my entire code is available on my first comment, above, if you would like to see it.

**Comment by flaviobvds** [comment]
> @flaviobvds does your app throws you a 504 error ? Does it affect GSSP too ?

Yes. I was getting the 504 timeout error too. When I digged further in, I found out the problem was that that `fetch` never ended.
I don't know if it affects GSSP because I don't have any on this project.
I have found some other [weird things](https://stackoverflow.com/questions/75752573/code-inside-then-after-sendgrid-request-runs-on-localhost-but-not-on-vercel) while on Vercel production too. Can't tell if they're connected somehow.

**Comment by Fasa2509** [comment]
I'm having the same problem. My app was working as expected three weeks ago, but then suddenly it just broke. Any api call to my own next endpoints takes +10s and returns 504 (and sometimes 502) errors, these calls also use a lot of memory, this includes too getServerSideProps.
I have tried to deploy in Netlify, use another Next version, fetching data with other frameworks, but nothing works.

I got this error message in my production logs

info  - Loaded env from /var/task/.env

**Comment by Fasa2509** [comment]
Bro, it doesn't make sense, BUT IT WORKED. Ctrl + Shift + F and I replaced at least 20 imports, completely worth it.
I'm really grateful with you, you have saved me. Thanks!!

**Comment by benevbright** [comment]
But WHY the front end code is included in the serverless bundle? 

> [API Routes](https://nextjs.org/docs/api-routes/introduction) are automatically optimized as isolated [Serverless Functions](https://vercel.com/features/infrastructure?utm_source=next-site&utm_medium=docs&utm_campaign=next-website) that can scale infinitely

![Screenshot 2023-04-11 at 21 21 48](https://user-images.githubusercontent.com/916690/231267332-7bb037e0-adbe-4ca3-842a-9a96f388a33b.png)

it does any optimisation?

**Comment by benevbright** [comment]
My best guess is that they have downgraded AWS lambda to the cheapest tier.

**Comment by tikhonov-aleksandr** [comment]
Hi guys, sorry for off topic, but maybe you know what dd1,dd2,dd3,dd4, ..., dd18 in the original json https://www.canada.ca/content/dam/ircc/documents/json/ee_rounds_123_en.json means? I want to parse this data, but have no idea what it is.
